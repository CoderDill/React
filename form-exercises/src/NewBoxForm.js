import { useState } from "react";
import { v4 as uuid } from "uuid";

const NewBoxForm = ({ addItem }) => {
  const [formData, setFormData] = useState({
    color: "",
    width: "",
    height: "",
  });
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addItem({ ...formData, id: uuid() });
    setFormData({
      color: "",
      width: "",
      height: "",
    });
  };
  return (
    <>
      <form onSubmit={handleSubmit} className="NewBoxForm">
        <label htmlFor="color">Add Color</label>
        <input
          name="color"
          type="text"
          placeholder="Add Color"
          value={formData.color}
          onChange={handleChange}
        />
        <label htmlFor="newBox">Add Width</label>
        <input
          name="width"
          type="text"
          placeholder="Add Width"
          value={formData.width}
          onChange={handleChange}
        />
        <label htmlFor="newBox">Add Height</label>
        <input
          name="height"
          type="text"
          placeholder="Add Height"
          value={formData.height}
          onChange={handleChange}
        />
        <button>Add Box</button>
      </form>
    </>
  );
};

export default NewBoxForm;
