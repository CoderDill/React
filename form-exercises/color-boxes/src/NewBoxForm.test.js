import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

it("should render", function () {
  render(<NewBoxForm />);
});

it("should match snapshot", function () {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});
