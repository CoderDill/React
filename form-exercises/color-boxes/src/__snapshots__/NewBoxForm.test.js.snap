// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="color"
    >
      Add Color
    </label>
    <input
      name="color"
      placeholder="Add Color"
      type="text"
      value=""
    />
    <label
      for="newBox"
    >
      Add Width
    </label>
    <input
      name="width"
      placeholder="Add Width"
      type="text"
      value=""
    />
    <label
      for="newBox"
    >
      Add Height
    </label>
    <input
      name="height"
      placeholder="Add Height"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
